addons/im_livechat/tests/test_get_random_operator.py

```diff
@@ -0,0 +1,91 @@
+import odoo
+from odoo.tests import HttpCase
+from odoo.tests.common import new_test_user
+from unittest.mock import patch
+from odoo.addons.bus.models.res_users import ResUsers
+
+def _compute_im_status(self):
+    for record in self:
+        record.im_status = 'online'
+
+
+@odoo.tests.tagged('-at_install', 'post_install')
+@patch.object(ResUsers, '_compute_im_status', _compute_im_status)
+class TestGetRandomOperator(HttpCase):
+    def _create_operator(self, lang_code=None, country_code=None):
+        operator = new_test_user(self.env, login=f'operator_{lang_code or country_code}_{self.operator_id}')
+        operator.partner_id = self.env['res.partner'].create({
+            'name': f'Operator {lang_code or country_code}',
+            'lang': lang_code,
+            'country_id': self.env['res.country'].search([('code', '=', country_code)]).id if country_code else None,
+        })
+        self.operator_id += 1
+        return operator
+
+    def setUp(self):
+        super().setUp()
+        self.operator_id = 0
+        self.env['res.lang'].with_context(active_test=False).search([
+            ('code', 'in', ['fr_FR', 'es_ES', 'de_DE', 'en_US'])
+        ]).write({'active': True})
+
+    def test_get_by_lang(self):
+        fr_operator = self._create_operator('fr_FR')
+        en_operator = self._create_operator('en_US')
+        livechat_channel = self.env['im_livechat.channel'].create({
+            'name': 'Livechat Channel',
+            'user_ids': [fr_operator.id, en_operator.id],
+        })
+        self.assertEqual(fr_operator, livechat_channel._get_random_operator(lang='fr_FR'))
+        self.assertEqual(en_operator, livechat_channel._get_random_operator(lang='en_US'))
```

Comment:
Each assertion on `_get_random_operator` has a 50% chance to be true, but not because it correctly selected the operator, be because you win the random lottery at https://github.com/odoo/odoo/pull/125911/files#diff-4c2954e7e011d14142d2c0ccae82ed8aa1c7aa5a111782442eee957b0801638dR283

We could isolate each selection logic inside a separate function and test only that. Yes it create 3 lines functions that feel useless but I don't see an other way to fix this.

Reply:
It's not a big deal. This assertions must always be true with the current setup. If the code breaks, then sure the test might work sometimes and fail sometimes, would be nicer that test successes or fails deterministically. However the functional specs are non-deterministically, so it's cheating if you change the internal logic of operator assignment.

For me test is good as it is. Better keep implementations very small and easy, and same for tests that match exactly the specs (tests are random if specs are random)